hostprocess/flannel/flanneld/start.ps1

```diff
@@ -32,13 +32,15 @@ Set-Content -Path $env:CNI_CONFIG_PATH/10-flannel.conf ($cniJson | ConvertTo-Jso
 Write-Host "add route"
 route /p add 169.254.169.254 mask 255.255.255.255 0.0.0.0
 
-write-host "copy sa info (should be able to do this with a change to go client"
-mkdir -force $env:CONTAINER_SANDBOX_MOUNT_POINT/flannel-config-file/var/run/secrets/kubernetes.io/serviceaccount/
-cp -force $env:CONTAINER_SANDBOX_MOUNT_POINT/var/run/secrets/kubernetes.io/serviceaccount/* $env:CONTAINER_SANDBOX_MOUNT_POINT/flannel-config-file/var/run/secrets/kubernetes.io/serviceaccount/
+# set host and port to Kubernetes API server
+$env:KUBERNETES_SERVICE_HOST = $env:CONTROL_PLANE_ENDPOINT.Split(':')[0]
```

Comment:
Is this going to override the ones created by kubernetes? https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables

Reply:
This is replacing these envs
https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#directly-accessing-the-rest-api

These envs are used by the in-cluster config. But instead going over the service created by Kubernetes we are directly communicating to the node (with it's actual IP)

Reply:
Sorry, I haven't had a chance to dig into this.  This seems strange to to overwrite what kubernetes is giving us.  I don't believe we needed to overwrite this information before.  I will try to get sometime soon to look more into this

Reply:
Oh well, I understand. I‘m sorry for bothering you
This is true. We weren‘t forced to overwrite this value before but the resulting behavior should be equal. Before we talked directly with one of the master nodes. With the overwrite we would instead talk over the service with one of the master node. This isn’t possible yet, though. Because the network isn’t ready for it yet. We discussed that a while back if you remember.
So instead of a larger change in flannel itself we are forcing Kubernetes (the pod) to talk to one of the master node directly by overwriting this value.

Don’t get me wrong. This isn’t the final solution. We’ll need that change in flannel. Especially if we want to integrate that more into the standard / kubeadm. But this isn’t possible by now

Though, the benefit for now is that we dropped dependencies to kubeadm-config and kube-proxy config 
In my opinion this is worth it. Also the needed change for us, once flannel made a change there, is way smaller if we merge this now. We simply update flannel and drop the overwriting :)

Reply:
not a bother! I really appreciate all the work you have done! I just haven't had chance to think deeply about it and with this type of change I want to make sure I fully understand.  The context you just added helps.

Reply:
ok, so it seems like we never ran into this because we _always_ had a kubeconfig with the management IP not service name (since kubeproxy has created all this). This was even the case when we installed flannel as a service on the host, not in a container.  I didn't look into the Linux side of things but they must be doing something (either in different order as you suggested or DNS is configured on the node?) different.  Since I don't think anyone's done this before it could be a bug in flannel as you suggested.

To unblock us, I think just setting the `KUBERNETES_SERVICE_HOST` env on the pod spec would be sufficient and not doing a conversion via script should work, could you try that?  

You have the open issue in flannel, might need someone to dive into Linux flannel pod and figure out what is going on there, so it could be solved long term.

Reply:
> [...] and not doing a conversion via script should work, could you try that?

Can you explain that a bit more? I'm unsure what you want me to do/try.

In case you mean that I should try to set the env inside of the yaml file: this could be problematic since I need to determine the value.

Reply:
> > [...] and not doing a conversion via script should work, could you try that?
> 
> Can you explain that a bit more? I'm unsure what you want me to do/try.
> 
> Do you mean that I should try to set the envs for host and port directly inside of the yaml? (why I didn't do this in the first place though 🤷‍♂️?)

yes, does that approach work?

Reply:
> > > [...] and not doing a conversion via script should work, could you try that?
> > 
> > 
> > Can you explain that a bit more? I'm unsure what you want me to do/try.
> > Do you mean that I should try to set the envs for host and port directly inside of the yaml? (why I didn't do this in the first place though 🤷‍♂️?)
> 
> yes, does that approach work?

That approach does indeed work, yes. Just tested and pushed the changes